components/inputs/input-date.js

```diff
@@ -109,21 +223,74 @@ class InputDate extends LocalizeStaticMixin(RtlMixin(LitElement)) {
 		if (elem) elem.focus();
 	}
 
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		changedProperties.forEach((oldVal, prop) => {
+			if (prop === 'value') {
+				if (this.value) this._formattedValue = formatISODateInUserCalDescriptor(this.value);
+				else this._formattedValue = '';
+
```

Comment:
Nit: might be more legible as:

    this.m_formattedValue = this.value ? formatISODateInUserCalDescriptor(this.value) : '';